dev/src/transaction.ts

```diff
@@ -771,3 +875,24 @@ function isRetryableTransactionError(error: GoogleError): boolean {
   }
   return false;
 }
+
+/**
+ * Delays further operations based on the provided error.
+ *
+ * @private
+ * @internal
+ * @return A Promise that resolves after the delay expired.
+ */
+async function maybeBackoff(
+  backoff: ExponentialBackoff,
+  error?: GoogleError
+): Promise<void> {
+  if ((error?.code as number | undefined) === StatusCode.RESOURCE_EXHAUSTED) {
+    backoff.resetToMax();
+  }
+  await backoff.backoffAndWait();
+}
+
+function resolveResult<TResult>({result}: {result: TResult}): TResult {
```

Comment:
This is a very small code snippet. I feel the indirection of calling this method obfuscates the code.

I think inlining this method with `v => v.result` is clearer and more concise.

Reply:
True